### ISSUE TITLE
Possible way around for 7.4+ versions

### ISSUE THREAD

**Issue by dezmen3** [Mixaill/FakePDB#29]
Played around with different versions of fakepdb and could finally make an .pdb file. I'm using IDA PRO 7.7

Steps:
1. Use fakepdb 0.3 to dump info into json file (might work with 0.2 dump, haven't tested)
2. Get fakepdb 0.2 and copy .json file and original exe file into fakepdb/win32/ folder or just take pdbgen.exe and place somewhere with those files
3. run cmd.exe->  pdbgen.exe generate -l "path to your app\app.exe" "path to json file\app.json" "path to .pdb file that will saved"

worked like a charm, got pdb,VS recognized functions

**Comment by widberg** [comment]
This works for me, thanks! Not sure what happened between 0.2 and 0.3 but using the older version I was able to get a PDB where the newer one would only produce a JSON. Having the same problem on the latest commit too.

**Comment by p0358** [comment]
I present to you the actual solution:

In IDA scripting window:
```py
import ida_nalt
ida_nalt.set_root_filename('E:\\new\\path\\to\\your\\analysedfile.dll')
```

You may also check current path with:
```py
ida_nalt.get_input_file_path()
```

This is the difference between 0.2 and 0.3, currently FakePDB uses `ida_nalt.get_input_file_path()`. Previously it assumed the old path to be `<ida_db_path>/<ida_db_name>.exe`. That assumption was of course very wrong, since you could have a different extension than exe, or even save the database elsewhere in the first place!

But the current behavior causes things to break if you move the executable somewhere else, while IDA still thinks it's at the old location. After changing it, it works again.

**Comment by privacyguy123** [comment]
This workaround works but it shouldn't eh? Why would the *older* version be better for *newer* IDA? I'm on 9.1 where things are especially broken/need fixes/workarounds.

**Comment by p0358** [comment]
@privacyguy123 how? I was just trying it out on 9.1, last week, and actually on Linux to make matters worse, and I didn't really seem to see any apparent issues on a first glance?

**Comment by privacyguy123** [comment]
> [@privacyguy123](https://github.com/privacyguy123) how? I was just trying it out on 9.1, last week, and actually on Linux to make matters worse, and I didn't really seem to see any apparent issues on a first glance?

How? I have no idea. Some major change from >v2.0 F'd something.

**Comment by sunbeam906** [comment]
> This workaround works but it shouldn't eh? Why would the _older_ version be better for _newer_ IDA? I'm on 9.1 where things are especially broken/need fixes/workarounds.

Can you share these workarounds? I am using 9.0 myself and for starters, it dies on "module 'ida_idaapi' has no attribute 'get_inf_structure'". Found that it was removed since IDA 8.

**Comment by p0358** [comment]
> I am using 9.0 myself and for starters, it dies on "module 'ida_idaapi' has no attribute 'get_inf_structure'". Found that it was removed since IDA 8.

Hmm... https://github.com/Mixaill/FakePDB/blob/23620cf28cabe935181d378c8950b272ae6e1bee/src_plugins/ida/fakepdb/dumpinfo.py#L678-L679

I think I know the problem now: you are probably using the last version from releases, which is 0.3 from 2021. You need to use the git master, there have been many commits since then that support IDA 9.x+ and a new release tag hasn't been made in that time.

**Comment by sunbeam906** [comment]
Yeah, I found some advice in here: https://github.com/Mixaill/FakePDB/issues/51#issuecomment-2863041689. So I've forked it > Actions > a Commit and now I'm waiting for the Workflow run to finish (10 minutes in). I wouldn't mind if you shared your build, a link to it, just in case - for some reason - I end-up with a fail :) Thank you!

**Comment by sunbeam906** [comment]
All good, build worked out ;)
